index.js

```diff
@@ -280,21 +288,26 @@ Client.prototype.getPaymentStatus = function(statusUrl, callback){
       } else {
         callback(response.body);
       }
-
   });
 };
 
 Client.prototype._getAndHandlePaymentStatus = function(statusUrl, callback, loopFunction){
   var self = this;
-  self.getPaymentStatus(statusUrl, function(error, response){
-    if(error){
-      callback(error, null);
-      return loopFunction(statusUrl, callback, loopFunction);
+  self.getPaymentStatus(statusUrl, function(error, response) {
+    if (error) {
+      if (error.response && error.response.body && error.response.body.error === 'txnNotFound') {
+        setTimeout(function() {
+          loopFunction(statusUrl, callback, loopFunction);
+        }, 100);
```

Comment:
Also, this will loop forever if the transaction is not found. This is probably in here since on initial submission ripple-rest didn't have the client_resource saved yet to its database. That should be fixed now.

Regardless it's good to be resilient to cases like that. Looping forever is not a good thing though. I would loop for a certain amount of time, maximum a minute. If the transaction didn't make it in by then it was lost somewhere.